File: babbleapp/src/main/java/com/learning/leap/bwb/settings/SettingOptionActivity.java
package com.learning.leap.bwb.settings;

import android.content.Intent;
import android.graphics.Bitmap;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import com.learning.leap.bwb.baseActivity.DetailActivity;
import com.learning.leap.bwb.R;
import com.learning.leap.bwb.utility.Utility;

public class SettingOptionActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setUpBackground();
        setContentView(R.layout.activity_setting_option);
        Button tipSettings = (Button)findViewById(R.id.settingsFragmentTipSettingButton);
        Button profileSettings = (Button)findViewById(R.id.settingsFragmentProfileSettingsButton);
        ImageView homeImageView = (ImageView)findViewById(R.id.settingsFragmentHomeImageView);
        ImageView playToday = (ImageView)findViewById(R.id.settingsFragmentPlayTodayImageView);
        ImageView libraryImageView = (ImageView)findViewById(R.id.settingsFragmentLibararyImageView);

        profileSettings.setOnClickListener(view -> userSettingsIntent());

        tipSettings.setOnClickListener(view -> tipsIntent());

        libraryImageView.setOnClickListener(view -> libraryIntent());

        playToday.setOnClickListener(view -> playTodayIntent());

        homeImageView.setOnClickListener(view -> homeIntent());

    }

    private void setUpBackground(){
        ImageView background = (ImageView)findViewById(R.id.tipSettingsBackground);
        Bitmap backgroundBitmap = Utility.decodeSampledBitmapFromResource(getResources(),R.drawable.settings_bg,Utility.getDisplayMetrics(this));
        background.setImageBitmap(backgroundBitmap);
    }

    private void userSettingsIntent(){
        Intent userSettingIntent = new Intent(SettingOptionActivity.this,UserInfoActivity.class);
        userSettingIntent.putExtra("newUser",false);
        startActivity(userSettingIntent);
    }

    private void tipsIntent(){
        Intent tipSettingsIntent = new Intent(SettingOptionActivity.this,TipSettingsActivity.class);
        startActivity(tipSettingsIntent);
    }

    private void libraryIntent(){
        Intent detailIntent = new Intent(SettingOptionActivity.this,DetailActivity.class);
        detailIntent.putExtra(DetailActivity.DETAIL_INTENT,DetailActivity.LIBRARY);
        startActivity(detailIntent);
    }

    private void playTodayIntent(){
        Intent detailIntent = new Intent(SettingOptionActivity.this,DetailActivity.class);
        detailIntent.putExtra(DetailActivity.DETAIL_INTENT,DetailActivity.PLAY_TODAY);
        startActivity(detailIntent);
    }

    private void homeIntent(){
        finish();
    }
}

File: babbleapp/src/main/java/com/learning/leap/bwb/settings/SettingOptionActivity.kt
package com.learning.leap.bwb.settings

import android.content.Intent
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.learning.leap.bwb.R
import com.learning.leap.bwb.baseActivity.DetailActivity
import com.learning.leap.bwb.settings.userInfo.UserInfoActivity
import kotlinx.android.synthetic.main.activity_setting_option.*

class SettingOptionActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting_option)
        setUpSettingRecyclerView()
        supportActionBar?.setBackgroundDrawable(ColorDrawable(ContextCompat.getColor(this,R.color.dark_green)))
        supportActionBar?.title = "Settings"
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

    }

    private fun setUpSettingRecyclerView(){
        val titles = resources.getStringArray(R.array.setting_titles)
        val detailStrings = resources.getStringArray(R.array.setting_detail)
        val adapter = SettingAdapter(titles.toList(),detailStrings.toList())
        settingsRecyclerView.adapter = adapter
        settingsRecyclerView.addItemDecoration(DividerItemDecoration(this, LinearLayoutManager.VERTICAL))
        settingsRecyclerView.layoutManager = LinearLayoutManager(this)
        adapter.itemOnClick = {
            when(it){
                0 -> userSettingsIntent()
                1 -> tipsIntent()
                else -> {
                ageRangeSettingsIntent()
                }
            }
        }

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val itemId: Int = item.itemId
        if (itemId == android.R.id.home) {
            finish()
        }
        return true

    }



    private fun userSettingsIntent() {
        val userSettingIntent = Intent(this@SettingOptionActivity, UserInfoActivity::class.java)
        userSettingIntent.putExtra("newUser", false)
        startActivity(userSettingIntent)
    }

    private fun ageRangeSettingsIntent() {
        val ageRangeIntent = Intent(this@SettingOptionActivity, AgeRangeActivity::class.java)
        startActivity(ageRangeIntent)
    }

    private fun tipsIntent() {
        val tipSettingsIntent = Intent(this@SettingOptionActivity, TipSettingsActivity::class.java)
        startActivity(tipSettingsIntent)
    }

    private fun libraryIntent() {
        val detailIntent = Intent(this@SettingOptionActivity, DetailActivity::class.java)
        detailIntent.putExtra(DetailActivity.DETAIL_INTENT, DetailActivity.LIBRARY)
        startActivity(detailIntent)
    }

    private fun playTodayIntent() {
        val detailIntent = Intent(this@SettingOptionActivity, DetailActivity::class.java)
        detailIntent.putExtra(DetailActivity.DETAIL_INTENT, DetailActivity.PLAY_TODAY)
        startActivity(detailIntent)
    }

    private fun homeIntent() {
        finish()
    }
}